artifacts/definitions/Windows/Analysis/EvidenceOfDownload.yaml

```diff
@@ -0,0 +1,35 @@
+name: Windows.Analysis.EvidenceOfDownload
+description: |
+   Simple artifact to find evidence of user download activity.
+   based on the Zone.Identifier alternate data stream that is created alongside with the file downloaded from the internet or intranet. Zone.Identifier is generated by applications when user saves files to the local file system from differnet security zone.
+   This artifact searches the directory provided for any file with alternate data stream named Zone.Identifier and then lists all files with zoneId = 3 or 4 and calculate the hash value of the file and prints the content of Zone.Identifier alternate stream as it could contain useful info in some cases.
+   
+
+reference:
+  - https://cyberforensicator.com/2018/06/26/where-did-it-come-from-forensic-analysis-of-zone-identifier/
+  - https://www.sans.org/security-resources/posters/windows-forensic-analysis/170/download
+   
+author: M.Soheem @msoheem
+
+# Can be CLIENT, CLIENT_EVENT, SERVER, SERVER_EVENT
+type: CLIENT
+
+parameters:
+   - name: DirectoryPath
+     default: "/C:/Users/dfir/Downloads/"
+
+sources:
+  - precondition:
+      SELECT OS From info() where OS = 'windows'
+
+    query: |
+     SELECT get(item=split(string=FullPath, sep=":Zone.Identifier"),member='0') as DownloadedFilePath,
+            hash(path=get(item=split(string=FullPath, sep=":Zone.Identifier"),member='0')) as FileHash,

```

Comment:
You can refactor it into two steps then you dont need to split twice:
```sql
LET X = SELECT split(string=FullPath, sep=":")[0] AS DownloadedFilePath, ... 
   FROM glob(...)

SELECT DownloadedFilePath, hash(path=DownloadedFilePath) AS FileHash
FROM X
```

Reply:
Sorry first artifact to write :)
Great tips Mike